packages/rocketchat-ldap/server/sync.js

```diff
@@ -84,14 +85,19 @@ getDataToSyncUserData = function getDataToSyncUserData(ldapUser, user) {
 					}
 					break;
 
-				case 'name':
+				default:
+					if (!_.find(whitelistedUserFields, (el) => el === userField.split('.')[0])) {
+						logger.debug(`user attribute not whitelisted: ${ userField }`);
+						return;
+					}
+
 					const tmpLdapField = RocketChat.templateVarHandler(ldapField, ldapUser.object);
+					const userFieldValue = _.reduce(userField.split('.'), (acc, el) => acc[el], user);
 
-					if (tmpLdapField && user.name !== tmpLdapField) {
-						userData.name = tmpLdapField;
-						logger.debug(`user.name changed to: ${ tmpLdapField }`);
+					if (tmpLdapField && userFieldValue !== tmpLdapField) {
+						userData[userField] = tmpLdapField;
```

Comment:
@goiaba Is not possible to have a field containing `.`, since you are creating an object like:
```
{
  "name.first": "Rodrigo"
}
```
and it should be 
```
{
  name: {
    first: 'rodrigo'
  }
}
```